app/containers/TextInput/FormTextInput.tsx

```diff
@@ -94,7 +95,15 @@ export default class FormTextInput extends React.PureComponent<IRCTextInputProps
 	}
 
 	get iconRight() {
-		const { iconRight, onIconRightPress, theme } = this.props;
+		const { iconRight, onIconRightPress, theme, onCancelSearch, value } = this.props;
+		if (onCancelSearch && value && value?.length > 0) {
```

Comment:
TextInput is supposed to be a generic component, so it doesn't make sense to have `onCancelSearch`.
This feature you're implementing here is to clear the input, so it should be `onClear` or `onClearInput`.